packages/cli/src/runner/file-command-runner.ts

```diff
@@ -2,36 +2,95 @@
 // Licensed under the MIT License.
 
 import { Spinner } from 'cli-spinner';
+import * as fs from 'fs';
 import { inject, injectable } from 'inversify';
+import * as lodash from 'lodash';
 import { ReportDiskWriter } from '../report/report-disk-writer';
 import { ReportGenerator } from '../report/report-generator';
+import { ConsoleSummaryReportGenerator } from '../report/summary-report/console-summary-report-generator';
+import { JsonSummaryReportGenerator } from '../report/summary-report/json-summary-report-generator';
+import { SummaryReportData } from '../report/summary-report/summary-report-data';
 import { AIScanner } from '../scanner/ai-scanner';
 import { AxeScanResults } from '../scanner/axe-scan-results';
 import { ScanArguments } from '../scanner/scan-arguments';
 import { CommandRunner } from './command-runner';
 
 @injectable()
 export class FileCommandRunner implements CommandRunner {
+    // tslint:disable-next-line: no-object-literal-type-assertion
+    private readonly summaryReportData = {
+        violationCountByRuleMap: {},
+        urlToReportMap: {},
+    } as SummaryReportData;
+
+    private readonly uniqueUrls = new Set();
+
     constructor(
         @inject(AIScanner) private readonly scanner: AIScanner,
         @inject(ReportGenerator) private readonly reportGenerator: ReportGenerator,
         @inject(ReportDiskWriter) private readonly reportDiskWriter: ReportDiskWriter,
+        @inject(JsonSummaryReportGenerator) private readonly jsonSummaryReportGenerator: JsonSummaryReportGenerator,
+        @inject(ConsoleSummaryReportGenerator) private readonly consoleSummaryReportGenerator: ConsoleSummaryReportGenerator,
+        private readonly fileSystemObj: typeof fs = fs,
+        private readonly lodashObj: typeof lodash = lodash,
```

Comment:
do we need this property for lodash? The import only should be sufficient

Reply:
we use this pattern in all our files so we can mock its functions in the uts

Reply:
Why we need to mock lodash functions? I do not see any code in project where we do mock it. We can mock fs that is okay, but mock lodash seems odd.

If we move to factory pattern for `AIScanner` instance creation we do not need to mock lodash.

Reply:
fs is mocked in service-configuration, axe-puppeteer, etc, isn't this the same pattern?, anyway that was removed